src/app/functions/server/getBreaches.ts

```diff
@@ -51,46 +51,53 @@ export async function getBreachIcons (breaches: Breach[]): Promise<LogoMap> {
     return new Map()
   }
   isFetchingIcons = true
-  const breachDomains = breaches
-    .map(breach => breach.Domain)
-    .filter(breachDomain => breachDomain.length > 0)
-  const logoFolder = pathResolve(dirname(fileURLToPath(import.meta.url)), '../../../../public/logo_cache/')
-  try {
-    await mkdir(logoFolder)
-  } catch {
-    // Do nothing; if the directory already exists, that's fine.
-  }
-  const existingLogos = await readdir(logoFolder)
-  // TODO: Batch to limit memory use?
-  const logoMapElems = await Promise.all(breachDomains.map(breachDomain => {
-    return new Promise((resolve, reject) => {
-      const logoFilename = breachDomain.toLowerCase() + '.ico'
-      const logoPath = pathResolve(logoFolder, logoFilename)
-      if (existingLogos.includes(logoFilename)) {
-        resolve([breachDomain, `/logo_cache/${logoFilename}`])
-        return
-      }
-      get(`https://icons.duckduckgo.com/ip3/${breachDomain}.ico`, (response) => {
-        if (response.statusCode !== 200) {
-          resolve(null)
+  async function fetchBreachIcons () {
+    const breachDomains = breaches
+      .map(breach => breach.Domain)
+      .filter(breachDomain => breachDomain.length > 0)
+    const logoFolder = pathResolve(dirname(fileURLToPath(import.meta.url)), '../../../../public/logo_cache/')
+    try {
+      await mkdir(logoFolder)
+    } catch {
+      // Do nothing; if the directory already exists, that's fine.
+    }
+    const existingLogos = await readdir(logoFolder)
+    // TODO: Batch to limit memory use?
+    const logoMapElems = await Promise.all(breachDomains.map(breachDomain => {
+      return new Promise((resolve, reject) => {
+        const logoFilename = breachDomain.toLowerCase() + '.ico'
+        const logoPath = pathResolve(logoFolder, logoFilename)
+        if (existingLogos.includes(logoFilename)) {
+          resolve([breachDomain, `/logo_cache/${logoFilename}`])
           return
         }
+        get(`https://icons.duckduckgo.com/ip3/${breachDomain}.ico`, (response) => {
+          if (response.statusCode !== 200) {
+            resolve(null)
+            return
+          }
 
-        const file = createWriteStream(logoPath)
-        response.pipe(file)
-        file.on('finish', () => {
-          file.close()
-          resolve([breachDomain, `/logo_cache/${breachDomain.toLowerCase()}.ico`])
+          const file = createWriteStream(logoPath)
+          response.pipe(file)
+          file.on('finish', () => {
+            file.close()
+            resolve([breachDomain, `/logo_cache/${breachDomain.toLowerCase()}.ico`])
```

Comment:
Re: `/logo_cache/${breachDomain.toLowerCase()}.ico`

Not sure if we can use `logoFilename` if it's still in scope from L68 above:

```suggestion
            resolve([breachDomain, `/logo_cache/${logoFilename}`])
```

Don't feel too strongly (and I know this is just tweaking of existing code) but one less place we're hardcoding filenames+extensions.

Reply:
Hmm yeah, we probably could, but again - avoiding yak shaving, unfortunately...